tests/cmds/test_show.py

```diff
@@ -0,0 +1,88 @@
+from chia.types.full_block import FullBlock
+from tests.cmds.cmd_test_utils import TestFullNodeRpcClient, TestRpcClients, cli_assert_shortcut, run_cli_command
+from tests.cmds.testing_classes import height_hash
+from tests.util.test_full_block_utils import get_full_blocks
+
+BASE_LIST = ["show"]
+
+
+@dataclass
+class ShowFullNodeRpcClient(TestFullNodeRpcClient):
+    full_block: FullBlock = field(init=False)
+
+    def __post_init__(self) -> None:
+        full_block: Optional[FullBlock] = None
+        for f_b in get_full_blocks(tx_block=True):
+            if f_b is not None and f_b.foliage_transaction_block is not None:
+                full_block = f_b
+                break
+        assert full_block is not None
+        self.full_block = full_block
+
+    async def get_fee_estimate(self, target_times: Optional[List[int]], cost: Optional[int]) -> Dict[str, Any]:
+        response: Dict[str, Any] = {
+            "current_fee_rate": 0,
+            "estimates": [0, 0, 0],
+            "fee_rate_last_block": 30769.681426718744,
+            "fees_last_block": 500000000000,
+            "full_node_synced": True,
+            "last_block_cost": 16249762,
+            "last_peak_timestamp": 1688858763,
+            "last_tx_block_height": 32,
+            "mempool_fees": 0,
+            "mempool_max_size": 0,
+            "mempool_size": 0,
+            "node_time_utc": 1689187617,
+            "num_spends": 0,
+            "peak_height": 12,
+            "success": True,
+            "target_times": target_times,
+        }
+        self.rpc_log["get_fee_estimate"] = (target_times, cost)
+        return response
+
+    async def get_block(self, header_hash: bytes32) -> Optional[FullBlock]:
+        # we return a block with height 2
+        self.rpc_log["get_block"] = (header_hash,)
+        return self.full_block
+
+
+RPC_CLIENT_TO_USE = ShowFullNodeRpcClient()  # pylint: disable=no-value-for-parameter
+
+
+def test_chia_show(capsys: Any, get_test_cli_clients: Tuple[TestRpcClients, Path]) -> None:
+    test_rpc_clients, root_dir = get_test_cli_clients
+    full_block = RPC_CLIENT_TO_USE.full_block  # a generated full block
+    # set RPC Client
+    test_rpc_clients.full_node_rpc_client = RPC_CLIENT_TO_USE
+    # get output with all options
+    command_args = ["-s", "-f", "-bh 1", "-b 0000000000000000000000000000000000000000000000000000000000000002"]
+    success, output = run_cli_command(capsys, root_dir, BASE_LIST + command_args)
+    assert success
+    tx_filter_hash: str = "Not a transaction block"
+    if full_block.foliage_transaction_block:
+        tx_filter_hash = full_block.foliage_transaction_block.filter_hash.hex()
+    # these are various things that should be in the output
+    assert_list = [
+        "Current Blockchain Status: Full Node Synced",
+        "Estimated network space: 25.647 EiB",
+        "Block fees: 500000000000 mojos",
+        "Fee rate:    3.077e+04 mojos per CLVM cost",
```

Comment:
this test is fragile. You generate a random FullBlock, tweaking the function generating that block, even in some small subtle way, will break this test. If many tests are like this, all tests would break.

It would be much better if this test was self-contained in that it specified the fields of the full block and then expected to get back those values here